pure-wasm/src/lib.rs

```diff
@@ -191,6 +191,22 @@ pub extern "C" fn take_state(engine_ptr: u32, json_ptr: u32, json_len: u32) {
     }
 }
 
+#[unsafe(no_mangle)]
+pub extern "C" fn get_state(engine_ptr: u32) -> u32 {
```

Comment:
I noticed we don't have any tests in pure-wasm and all functions are tested in language bindings. I will do the same for this one

Reply:
Yeah, this is kinda intentional. Most of this is wrapping and the only real way to test this is to setup a Rust FFI reader for this and poke it in the same way the host would poke it. Which feels daft because we can just test this with host tests. I'm open to being told I'm crazy here and we can always set something up if we need